alexandria/core/views.py

```diff
@@ -139,6 +139,27 @@ def multi(self, request, **kwargs):
 
             return response
 
+    @action(methods=["post"], detail=True, url_path="generate-thumbnail")
+    def generate_thumbnail(self, request, pk=None):
+        if not settings.ALEXANDRIA_ENABLE_THUMBNAIL_GENERATION:
+            raise ValidationError(_("Thumbnail generation is not enabled."))
+
+        file = self.get_object()
+        if file.variant == models.File.THUMBNAIL:
+            raise ValidationError(
+                _("File is already a thumbnail, cannot generate thumbnail.")
+            )
+        elif file.renderings.all().count() > 0:
```

Comment:
```suggestion
        elif file.renderings.all().exists():
```

This generates better/more performant SQL

Reply:
Done